### DISCUSSION TITLE
Code actions sorting

### DISCUSSION THREAD

**Discussion by davidmh** [nvimtools/none-ls.nvim#62] [Ideas]
Currently, all code actions are sorted by their title:

https://github.com/nvimtools/none-ls.nvim/blob/cf81e765460f722a381e5f3f5e3cbdce88398517/lua/null-ls/code-actions.lua#L38-L41

But not all code action sources want an alphabetical order.

For example, in the [CSpell](/davidmh/cspell.nvim)  code actions, I would like to show fixes, suggestions, and actions to add exceptions in that order.

The source already inserts the actions in the desired order, but as soon as we pass the action to none-ls, the table gets sorted by title, which is different from what the source intended.

How do we feel about just removing the sorting? Alternatively, we could update the generator API to read a sorting function for the code actions.

Thoughts?

**Comment by mochaaP** [comment]
Yeah, I think removing it makes more sense.

**Reply by davidmh** [reply]
Cool, I'll send a PR shortly
